holoviews/core/dimension.py

```diff
@@ -116,57 +154,120 @@ class Dimension(param.Parameterized):
-            existing_params = dict(self.presets[(str(name),)].get_param_values())
+        elif spec in self.presets:
+            existing_params = dict(self.presets[spec].get_param_values())
+        elif (spec,) in self.presets:
+            existing_params = dict(self.presets[(spec,)].get_param_values())
         else:
-            existing_params = {'name': name}
+            existing_params = {}
 
         all_params = dict(existing_params, **params)
-        name = all_params['name']
-        label = name
-        if isinstance(name, tuple):
-            name, label = name
+        if isinstance(spec, tuple):
+            name, label = spec
             all_params['name'] = name
-        self.label = label
-
-        if not isinstance(params.get('values', None), basestring):
-            all_params['values'] = sorted(list(unique_array(params.get('values', []))))
-        elif params['values'] != 'initial':
-            raise Exception("Values argument can only be set with the string 'initial'.")
+            all_params['label'] = label
+            if 'label' in params and (label != params['label']):
+                self.warning('Using label as supplied by keyword ({!r}), ignoring '
+                             'tuple value {!r}'.format(params['label'], label))
+                all_params['label'] = params['label']
+        elif isinstance(spec, basestring):
+            all_params['name'] = spec
+            all_params['label'] = params.get('label', spec)
+
+        if all_params['name'] == '':
+            raise ValueError('Dimension name cannot be the empty string')
+        if all_params['label'] in ['', None]:
+            raise ValueError('Dimension label cannot be None or the empty string')
+
+        values = params.get('values', [])
+        if isinstance(values, basestring) and values == 'initial':
+            self.warning("The 'initial' string for dimension values is no longer supported.")
+            values = []
+
+        all_params['values'] = sorted(list(unique_array(values)))
         super(Dimension, self).__init__(**all_params)
 
 
-    def __call__(self, name=None, **overrides):
+    @property
+    def spec(self):
+        "Returns the corresponding tuple specification"
+        return (self.name, self.label)
+
+
+    def __call__(self, spec=None, **overrides):
+        "Aliased to clone method. To be deprecated in 2.0"
+        return self.clone(spec=spec, **overrides)
+
+
+    def clone(self, spec=None, **overrides):
         """
         Derive a new Dimension that inherits existing parameters
         except for the supplied, explicit overrides
         """
         settings = dict(self.get_param_values(onlychanged=True), **overrides)
-        if name is not None: settings['name'] = name
-        return self.__class__(**settings)
+        spec = spec if (spec is not None) else (overrides.get('name', self.name),
+                                                overrides.get('label', self.label))
```

Comment:
Does it matter? If there is a spec, it gets passed through as the first argument, shouldn't matter if it is just a string or just a tuple.